lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -128,10 +152,230 @@ hashSeed :: ScrubbedBytes -> ScrubbedBytes
+|                                                                                |
+|      DdzFFzCqrhstiVdBdYEAmpLPtSWxFYy...rYcBLq29xJD4xZw16REKyhJC9PFGgPSbX       |
+|                                                                                |
++--------------------------------------------------------------------------------+
+                                    |
+                                    |
+                                    v
++--------------------------------------------------------------------------------+
+|     Address Root    |     Address Attributes    |           AddrType           |
+|                     |                           |                              |
+|   Hash (224 bits)   |  Der. Path* + Stake + NM  |  PubKey | (Script) | Redeem  |
+|                     |    (open for extension)   |     (open for extension)     |
++--------------------------------------------------------------------------------+
+             |                 |
+             |                 |     +----------------------------------+
+             v                 |     |        Derivation Path           |
++---------------------------+  |---->|                                  |
+|  SHA3-256 >> Blake2b_224  |  |     | ChaChaPoly* AccountIx/AddressIx  |
+|                           |  |     +----------------------------------+
+|  -AddrType                |  |
+|  -ASD* (~AddrType+PubKey) |  |     +----------------------------------+
+|  -Address Attributes      |  |     |       Stake Distribution         |
++---------------------------+  |     |                                  |
+                               |---->|  BootstrapEra | (Single | Multi) |
+                               |     +----------------------------------+
+                               |
+                               |     +----------------------------------+
+                               |     |          Network Magic           |
+                               |---->|                                  |
+                                     | Addr Discr: MainNet vs TestNet   |
+                                     +----------------------------------+
+
+-------------------------------------------------------------------------------}
+
+{-
+-- The full address generation function, for reference.
+rndToAddress
+    :: Key 'AddressK XPub
+    -> Key 'RootK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+rndToAddress (Key addrXPub) rootKey accIx addrIx =
+    Address $ CBOR.toStrictByteString $
+    encodeAddress addrXPub $
+    encodeAddressAttributes rootKey accIx addrIx
+
+encodeAddressAttributes
+    :: Key 'RootK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> CBOR.Encoding
+encodeAddressAttributes rootKey accIx addrIx = mempty
+    <> CBOR.encodeMapLen 1
+    <> CBOR.encodeWord8 1
+    <> encodeDerivationPath rootKey accIx addrIx
```

Comment:
Note that this is only true for `mainnet` addresses. On `testnet`, address attributes also contains an extra field for the network discrimination (cf: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/http-bridge/src/Cardano/Wallet/HttpBridge/Compatibility.hs#L105-L128).